golang/internal/mapper/grpc.go

```diff
@@ -514,9 +526,19 @@ func mapServicePorts(svc *corev1.Service) []*common.ContainerStateItemPort {
 }
 
 // do better mapping this is quick something
-func mapKubeStatusToCruxContainerState(status appsv1.DeploymentStatus) common.ContainerState {
-	if status.ReadyReplicas > 0 {
+func mapKubeStatusToCruxContainerState(state corev1.ContainerState) common.ContainerState {
+	if state.Running != nil {
 		return common.ContainerState_RUNNING
 	}
+	if state.Terminated != nil {
+		return common.ContainerState_EXITED
+	}
+	if state.Waiting != nil {
+		if state.Waiting.Reason == "CrashLoopBackOff" {
```

Comment:
Which kubernetes statuses should we map? There are lots of them.
Or should we send the state as a string if it's unknown?

Reply:
I thought we would map the Kubernetes statuses, we should send as specific an error as we can, if we map it to our custom statuses (which are docker related) than it can confuse the users. 

https://github.com/kubernetes/kubernetes/blob/6356023cb42d681b7ad0e6d14d1652247d75b797/pkg/kubelet/container/sync_result.go#L32